ctrd/utils.go

```diff
@@ -166,3 +169,31 @@ func toLinuxResources(resources types.Resources) (*specs.LinuxResources, error)
 
 	return r, nil
 }
+
+// convertCtrdErr converts containerd client error into a pouchd manager error.
+// containerd client error converts GRPC code from containerd API to containerd client error.
+// pouchd manager error is used in the whole managers and API layers to construct status code for API.
+// there should be a way convert the previous to the latter one.
+func convertCtrdErr(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	if errdefs.IsNotFound(err) {
+		return errors.Wrap(errtypes.ErrNotfound, err.Error())
+	}
+
+	if errdefs.IsAlreadyExists(err) {
+		return errors.Wrap(errtypes.ErrAlreadyExisted, err.Error())
```

Comment:
it;s ok to have duplicate error message. we might find the way to hack it and remove the duplicated message